src/components/tools/table-tool/cell-formatter.tsx

```diff
@@ -1,14 +1,33 @@
-import { TFormatterProps } from "./table-types";
+import React from "react";
+import { kCellHorizontalPadding, kCellLineHeight, kDefaultColumnWidth, kRowHeight, TRow } from "./table-types";
 import { useNumberFormat } from "./use-number-format";
 
-export const formatValue = (formatter: (n: number | { valueOf(): number }) => string, value: any) => {
-  if ((value == null) || (value === "")) return "";
+import './cell-formatter.scss';
+
+export const formatValue = (
+    formatter: (n: number | { valueOf(): number }) => string, value: any,
+    width?: number, row?: TRow, rowHeight?: (args: any) => number
+  ) => {
+  if ((value == null) || (value === "")) return <span></span>;
   const num = Number(value);
-  if (!isFinite(num)) return value;
-  return formatter(num);
+  if (!isFinite(num)) {
+    const cellWidth = (width || kDefaultColumnWidth) - kCellHorizontalPadding;
+    const height = rowHeight && row ? rowHeight({ row }) : kRowHeight;
+    return (
+      <div className="text-cell" style={{ height, width: cellWidth, lineHeight: `${kCellLineHeight}px` }}>
+        {value}
+      </div>
+    );
+  }
+  return <span>{formatter(num)}</span>;
 };
 
-export const CellFormatter: React.FC<TFormatterProps> = ({ column, row }) => {
-  const formatter = useNumberFormat();
-  return formatValue(formatter, row[column.key]);
+export const getCellFormatter = (width: number, rowHeight: (args: any) => number) => {
+  // args.row: TRow
+  // args.column: CalculatedColumn<TRow, any>
+  return (args: any) => {
```

Comment:
Having this as a standalone component makes it stable, i.e. so it doesn't cause table rerenders when it changes. This is going to return a new component instance on every call, which is then going to result in a change to the `columns`, which will re-render the entire table. From the RDG docs:

>⚠️ Passing a new columns array will trigger a re-render for the whole grid, avoid changing it as much as possible for optimal performance.

In theory, one could cache all of the formatters returned from this function, but that could be a lot of formatter instances. The column width is available from the column, which is passed in as an argument, so that doesn't need to be a parameter. The `rowHeight()` function is stable, so it's not going to invalidate the columns, but it's still being called in the formatting of every cell which is inefficient. There are a couple of alternatives for handling the `rowHeight`:

1. It could be added to the `TRow` structure (e.g. as `__height__`), in which case it could be computed in `useRowsFromDataSet()` and then looked up from the `row` argument here.
2. Another approach, which is the way row selection is now handled, is that we could override the `RowRenderer` to put the `rowHeight` in context and then the cells could access it via a `useRowHeightContext()` hook.

But these would fall in the category of performance improvements that probably aren't worth it right now.